ccd-definitions/src/main/java/uk/gov/hmcts/reform/divorce/ccd/builder/FieldCollectionBuilderExtension.java

```diff
@@ -0,0 +1,130 @@
+package uk.gov.hmcts.reform.divorce.ccd.builder;
+
+import de.cronn.reflection.util.TypedPropertyGetter;
+import lombok.Getter;
+import uk.gov.hmcts.ccd.sdk.api.Event.EventBuilder;
+import uk.gov.hmcts.ccd.sdk.api.FieldCollection.FieldCollectionBuilder;
+import uk.gov.hmcts.reform.divorce.ccd.model.CaseData;
+import uk.gov.hmcts.reform.divorce.ccd.model.State;
+import uk.gov.hmcts.reform.divorce.ccd.model.UserRole;
+
+import static uk.gov.hmcts.ccd.sdk.api.DisplayContext.Mandatory;
+
+@Getter
+public class FieldCollectionBuilderExtension {
+
+    private final FieldCollectionBuilder<CaseData, EventBuilder<CaseData, UserRole, State>> fieldCollectionBuilder;
+
+    public FieldCollectionBuilderExtension(
+        final FieldCollectionBuilder<CaseData, EventBuilder<CaseData, UserRole, State>> fieldCollectionBuilder) {
+        this.fieldCollectionBuilder = fieldCollectionBuilder;
+    }
+
+    public FieldCollectionBuilderExtension page(final String id) {
```

Comment:
Maybe this would be less repetitive through inheritance?

Reply:
Moved to ccd generator builder